### PR TITLE
(FACT-3142) Add support for Mariner to Facter 4.x

### PR DESCRIPTION
**Pull request by mminkus**
Add support for CBL-Mariner, Microsoft's Linux distribution for
cloud infrastructure and edge products and services.

### CODE REVIEW

File: lib/facter/resolvers/os_release.rb [hunk 1]
```diff
@@ -85,6 +85,8 @@ def join_os_name
           @fact_list[:name] = if os_name.downcase.start_with?('red', 'oracle', 'arch', 'manjaro')
                                 os_name = os_name.split(' ')[0..1].join
                                 os_name
+                              elsif os_name.downcase.end_with?('mariner')
+                                os_name = os_name.split(' ')[-1].strip
```
Comment (tvpartytonight) [inline lib/facter/resolvers/os_release.rb:89]: To make rubocop happy, I think you will need to explicitly return `os_name` as is done in the previous condition, or not assign it at all, in the subsequent condtion.

Comment (mminkus) [reply lib/facter/resolvers/os_release.rb:89]: You are correct, confirmed removing the assignment causes it to pass rubocop.

Comment (mminkus) [reply lib/facter/resolvers/os_release.rb]: Pushed fix, it passes rubocop and rspec.

**Comment by joshcooper** [comment]
@mminkus could you squash your commits and prefix the commit summary with `(FACT-3142)`?

**Comment by mminkus** [comment]
> @mminkus could you squash your commits and prefix the commit summary with `(FACT-3142)`?

I thought you can squash the commits when they are merged. Or do you want me to squash the commits locally and do a force push?

**Comment by joshcooper** [comment]
Please squash and amend the commit summary (as our automation links commits to JIRA tickets based on the summary line) and force push to your branch. Thanks!

**Review by tvpartytonight — APPROVED** [review]

**Pull request merged**